src/dc_msg.rs

```diff
@@ -786,35 +786,29 @@ pub unsafe fn dc_msg_get_summary<'a>(
     msg: *mut dc_msg_t<'a>,
     mut chat: *const Chat<'a>,
 ) -> *mut dc_lot_t {
-    let current_block: u64;
+    let mut chat_exists = true;
```

Comment:
should init to false to be on the safe side

Reply:
The function is small enough to refactor into something rusty. In this case, an Option can be used.

Reply:
I think initing to false should be done in a refactoring. In this case we're just making the goto logic in something more readable, the logic stays the same. So it feels more safe, but isn't. Especially because we need to reintroduce all those else arms which do nothing besides setting the variable to true.

Reply:
maybe it's best to consistently name things OK_TO_CONTINUE (default to true) and then count that in top_evil.py with the goal to get zero occurences.